src/utils/adopter.js

```diff
@@ -42,7 +42,7 @@ export function makeInstance (element, isHTML = false) {
 }
 
 export function nodeOrNew (name, node) {
-  return node instanceof globals.window.Node ? node : create(name)
+  return (node && 'ownerDocument' in node && node instanceof node.ownerDocument.defaultView.Node) ? node : create(name)
```

Comment:
probably checking `node.ownerDocument` is better here

Reply:
oh right. you got me thinking what is the difference anyway it seems to be that 

> [If an object has a property whose value is undefined, then comparing the property with undefined will return an incorrect result.](https://www.javascripttutorial.net/object/javascript-check-if-property-exists/)

But when checking node.ownerDocument there seems to be no problems and the tests pass just the same so I agree.

Reply:
Well in javascript `undefined` usually means "not defined" on the object. Even if you set the property to undefined, that's what you often want to communicate. Otherwise, you use `null`. The `in` check is an escape hatch in case you REALLY need it. Usually checking just for a truthy value is enough.
I also realized that we don't really need the `instanceof` check anymore because the whole purpose of this check is to distinguish a Node from a plain object.
But I guess its fine like that. Thanks!

Reply:
Okay. I was thinking the same thing about using `instanceof` actually.  The main reason I left it was because it was what was originally being done.

Reply:
We'll leave it for now :)